samcli/cli/cli_config_file.py

```diff
@@ -57,14 +57,12 @@ def __call__(self, config_path, config_env, cmd_names):
 
         samconfig = SamConfig(config_file_dir, config_file_name)
 
-        # Enable debug level logging by environment variable "SAM_DEBUG"
-        if os.environ.get("SAM_DEBUG", "").lower() == "true":
```

Comment:
It never worked unless SAM_DEBUG was set manually apart from --debug flag because click doesn't create/set environment variables

Reply:
`--debug` also loads from this envvar. In case this callback is executed before callback of `--debug`, the log level will not be properly set and here the debug log will not be shown.

Reply:
It used to be the case but now it is not. Now `--debug` callback is executed before. I kept [SAM_DEBUG](https://github.com/aws/aws-sam-cli/blob/4e1a9796d9538e37dacc170d550c4f5561e10f63/samcli/cli/options.py#L27) as a back door though. Hopefully, we won't ever need it because it makes UX confusing